src/CanvasRenderingContext2d.cc

```diff
@@ -2451,8 +2451,19 @@ NAN_SETTER(Context2d::SetFont) {
 
   const int argc = 1;
   Local<Value> argv[argc] = { value };
-  Local<Value> parsed = _parseFont.Get(iso)->Call(ctx, ctx->Global(), argc, argv).ToLocalChecked();
-  Local<Object> font = Nan::To<Object>(parsed).ToLocalChecked();
+
+  Nan::TryCatch try_catch;
+  MaybeLocal<Value> maybeParsed = _parseFont.Get(iso)->Call(ctx, ctx->Global(), argc, argv);
+  if (try_catch.HasCaught()){
+    try_catch.ReThrow();
+    return;
+  }
+  MaybeLocal<Object> maybeFont = Nan::To<Object>(maybeParsed.ToLocalChecked());
+  if (try_catch.HasCaught()){
+    try_catch.ReThrow();
+    return;
+  }
+  Local<Object> font = maybeFont.ToLocalChecked();
```

Comment:
Thanks for the fast fix :).

A bit simpler: you can explicitly check if the `MaybeLocal` is empty with something like:

```cpp
  MaybeLocal<Object> mlfont = Nan::To<Object>(parsed);
  if (mlfont.IsEmpty()) return;
  Local<Object> font = mlfont.ToLocalChecked();
```

The error will still be thrown appropriately when the handle is empty.

Reply:
Done. Is there a need to test if `mparsed` is empty (can `parseFont` itself throw an error)?

Reply:
Technically I'm not sure if there's any v8 operation that returns a `MaybeLocal` that is guaranteed not to fail. I'm not sure how unwise it is to assume simple operations don't fail. Having it doesn't hurt.